services/pane.js

```diff
@@ -0,0 +1,125 @@
+var _ = require('lodash'),
+  dom = require('./dom'),
+  ds = require('dollar-slice'),
+  paneController = require('../controllers/pane'),
+  publishPaneController = require('../controllers/publish-pane');
+
+/**
+ * create pane
+ * @param {string} header
+ * @param {Element|DocumentFragment} innerEl
+ * @returns {Element}
+ */
+function createPane(header, innerEl) {
+  var template = dom.find('.kiln-pane-template'),
+    el = document.importNode(template.content, true);
+
+  // add header and contents
+  el.querySelector('.pane-header').innerHTML = header;
+  el.querySelector('.pane-inner').appendChild(innerEl);
+
+  return el;
+}
+
+/**
+ * close an open pane
+ */
+function close() {
+  var pane  = dom.find('.kiln-toolbar-pane-background');
+
+  if (pane) {
+    dom.removeElement(pane);
+  }
+}
+
+/**
+ * open a pane
+ * @param {string} header
+ * @param {Element} innerEl
+ * @returns {Element} pane
+ */
+function open(header, innerEl) {
+  var toolbar = dom.find('.kiln-toolbar'),
+    el = createPane(header, innerEl),
+    pane;
+
+  close(); // close any other panes before opening a new one
+  dom.insertBefore(toolbar, el);
+  pane = toolbar.previousElementSibling; // now grab a reference to the dom
+  // init controller for pane background
+  ds.controller('pane', paneController);
+  ds.get('pane', pane);
+  return pane;
+}
+
+/**
+ * open publish pane
+ */
+function openPublish() {
+  var header = 'Schedule Publish',
+    actionsEl = dom.create(`
+      <div class="actions">
+        <button class="publish-now">Publish Now</button>
+      </div>
+    `), // todo: add other publishing actions, depending on page state
+    // e.g. schedule publish, message, cancel schedule, unpublish
+    el = open(header, actionsEl);
+
+  // init controller for publish pane
+  ds.controller('publish-pane', publishPaneController);
+  ds.get('publish-pane', el.querySelector('.actions'));
+}
+
+function addPreview(preview) {
+  if (preview) {
+    return `<span class="error-preview">${preview}</span>`;
+  } else {
+    return '';
+  }
+}
+
+function addErrors(errors) {
```

Comment:
Looks like errors array requires objects with certain properties -- need to add to JSDoc.